plugins/github-pull-requests-board/src/hooks/usePullRequestsByTeam.tsx

```diff
@@ -43,18 +47,52 @@ export function usePullRequestsByTeam(
 
       return pullRequestsWithDetails;
     },
-    [getPullRequests, getPullRequestDetails, pullRequestLimit],
+    [getPullRequestsFromRepository, getPullRequestDetails, pullRequestLimit],
+  );
+
+  const getPRsPerTeamMember = useCallback(
+    async (
+      teamMember: string,
+      teamOrganization?: string,
+    ): Promise<PullRequests> => {
+      const pullRequestsNumbers = await getPullRequestsFromUser(
+        teamMember,
+        teamOrganization,
+        pullRequestLimit,
+      );
+
+      const pullRequestsWithDetails = await Promise.all(
+        pullRequestsNumbers.map(({ node }) =>
+          getPullRequestDetails(
+            `${node.repository.owner.login}/${node.repository.name}`,
+            node.number,
+          ),
+        ),
+      );
+
+      return pullRequestsWithDetails;
+    },
+    [getPullRequestsFromUser, getPullRequestDetails, pullRequestLimit],
   );
 
   const getPRsFromTeam = useCallback(
-    async (teamRepositories: string[]): Promise<PullRequests> => {
+    async (
+      teamRepositories: string[],
+      teamMembers: string[],
+      teamOrganization?: string,
+    ): Promise<PullRequests> => {
       const teamRepositoriesPromises = teamRepositories.map(repository =>
         getPRsPerRepository(repository),
       );
 
-      const teamPullRequests = await Promise.allSettled(
-        teamRepositoriesPromises,
-      ).then(promises =>
+      const teamMembersPromises = teamMembers.map(teamMember =>
```

Comment:
same here with limits

Reply:
sooo, this one follows the implementation of `useGetPullRequestsFromRepository` which has some limit logic baked in to set a limit of PRs **per repository**. This means that this new `usePullRequestsByTeam` also has some limit logic **per user** (e.g.: max 100 PRs per user).

Now, I think this limit logic has a major limitation since it will blow up if the team has a huge amount of repos or a huge amount of members (e.g.: let's say an organization that puts everyone in the same team and has many teams/repos). I'm not sure of the likelihood though.

I guess some ways to approach it:
* Set a global limit of PRs that applies across all repos and users
* And/Or set a limit of repos/users
* And/Or only pull the member repos on-demand, when the user actually clicks the tab

Curious to have your thoughts on that before I implement some changes 🤔

Reply:
I think if it makes things easier, it's probably OK to not worry too much about the global picture for now, but instead just making each little local piece have p-limit of like three around it. Doesn't make a lot of sense to have higher concurrency than that anyway, you'll just get rate limited.

If the p-limit is created at the root of this file, then it'll be effectively global anyway. As in, if there's a hundred concurrent calls to `getPRsFromTeam`, then they'll get the internal work queued up on that global queue anyway and all wait for each other.

Reply:
As in,

```ts
import limiterFactory from 'p-limit';
const teamPrsLimiter = limiterFactory(3);
// ...
async function fnThatIsCalledWithHugeConcurrency() {
  await Promise.all(something.map(s => limiter(async () => {
    // THIS particular code block can never, under any circumstances,
    // run with a higher concurrency than 3, globally
  }));
}
```

Reply:
interesting, just learned something new, I'll give it a try

Reply:
Okay, I've been able to add some p-limit as you suggested, and that seems to work. I've set the limit to 5 for now, rather than 3, otherwise the page can be a bit slow to load.

One small thing though: I wasn't able to just declare the p-limit at the top, and then use it wherever I'm doing my async call. The linter complained that I wasn't adding the p-limit to the useCallback dependencies, but doing so caused the page to completely freeze. I'm not familiar enough with React/Typescript to know what's the best way forward here, so I've just used separate p-limit for `getPRsPerRepository` and `getPRsPerTeamMember` and that seems to work fine 🤔